sycl/doc/extensions/experimental/sycl_ext_oneapi_graph.asciidoc

```diff
@@ -280,12 +280,13 @@ dependencies in one of three ways. Firstly, through buffer accessors that
 represent data dependencies between two command groups captured as nodes.
 Secondly, by using the `handler::depends_on()` mechanism inside a command group
 captured as a node. However, for an event passed to `handler::depends_on()` to
-create an edge, it must be an event returned from a queue
-submission captured by the same graph. Otherwise, a synchronous error will be
-thrown with error code `invalid`. `handler::depends_on()` can be
-used to express edges when a user is working with USM memory rather than SYCL
-buffers. Thirdly, for a graph recorded with an in-order queue, an edge is added
-automatically between two sequential command groups submitted to the in-order queue.
+create an edge, it must be an event returned from a queue submission captured by
+the same graph. Passing events from other sources (other graph submissions,
```

Comment:
maybe we can list all the sources for the event at some place in the spec. and also give a naming to each event type, and then the spec can be described more clear.

regular sycl submission
sycl graph submission
sycl submission during recording
query a given node of an exec graph
...

Reply:
We previously had such a description in the spec only to help with understanding the differences, but to my mind there are only two relevant types:

- Events which come from a recorded graph submission (previously described as "Graph-limited events"
- All other events should be considered normal SYCL events for all intents and purposes

@gmlueck Do y ou have any objections to bringing back some kind of definition like this in the spec only to help with clarity? It is a bit verbose having to describe the type of event every single time it is mentioned like in the example above.